src/client/contexts/GoogleAnalyticsContext.tsx

```diff
@@ -0,0 +1,76 @@
+import React, {
+  FC,
+  createContext,
+  useContext,
+  useEffect,
+  useState,
+} from 'react'
+import { useLocation } from 'react-router-dom'
+import {
+  GA_USER_EVENTS,
+  initializeGA,
+  setGAUserId,
+  sendPageView,
+  sendUserEvent,
+  sendTiming,
+  sendException,
+} from '../services'
+
+interface GoogleAnalyticsContextProps {
+  GA_USER_EVENTS: {
+    SUBMIT: string
+  }
+  setGAUserId: (userId: number | null) => void
+  sendPageView: (path: string) => void
+  sendUserEvent: (action: string, label?: string, value?: number) => void
+  sendTiming: (category: string, variable: string, value: number) => void
+  sendException: (description: string) => void
+}
+
+export const GoogleAnalyticsContext = createContext<
+  GoogleAnalyticsContextProps | undefined
+>(undefined)
+
+export const useGoogleAnalytics = (): GoogleAnalyticsContextProps => {
+  const GoogleAnalytics = useContext(GoogleAnalyticsContext)
+  if (!GoogleAnalytics)
+    throw new Error(
+      'useGoogleAnalytics must be used within an GoogleAnalyticsProvider'
+    )
+  return GoogleAnalytics
+}
+
+export const GoogleAnalyticsProvider: FC = ({ children }) => {
+  const [isLoaded, setLoaded] = useState(false)
+
+  const location = useLocation()
+
+  useEffect(() => {
+    try {
+      initializeGA()
+    } catch (err) {
```

Comment:
Not sure if a try-catch is needed here. Copied this from: https://github.com/opengovsg/postmangovsg/pull/285/files#diff-9893cc2cc2b8bcca8c9660d02fc3422acf19cc738bb01fba2a1afcfc69c9f970L30-L32